packages/generator/src/edmx-to-vdm/v4/operation.ts

```diff
@@ -120,17 +120,31 @@ export function filterAndTransformOperations(
   operations: EdmxOperation[],
   isBound: boolean
 ): EdmxJoinedOperation[] {
-  const filteredByBoundOperations = operations.filter(
-    operation => (operation.IsBound.toLowerCase() === 'true') === isBound
+  const [withoutOperations, withOperations] = splitOperationsByMissingImports(
+    operationImports,
+    operations
   );
 
-  const [withoutOperation, withOperation] = splitMissingOperation(
-    operationImports,
-    filteredByBoundOperations
+  const relevantOperations = withOperations.filter(
+    operation => operation.IsBound === isBound
   );
 
+  const validBoundOperations =
+    validateBoundOperationParameters(relevantOperations);
+
+  validateOperationImports(
+    withoutOperations,
+    withoutOperations[0]?.operationType
+  );
```

Comment:
[nth] If you move this up this whole section gets a little easier to implement. Also check the comments above and below for details.

```
  validateOperationImports(
    withoutOperations,
    withoutOperations[0]?.operationType
  );

  if (isBound) {
    return validateBoundOperationParameters(relevantOperations);
  }

  return relevantOperations;
```